### PR TITLE
[CN-672] Add page with Native Memory parameters

### PR DESCRIPTION
**Pull request by dzeromski-hazelcast**

### CODE REVIEW

**Review by cagric0 — APPROVED** [review]
LGTM, but I think you should also get a review from Angela.

**Review by hasancelik — APPROVED** [review]
LGTM 👍 as @cagric0 said, getting reviews from doc team would be good before the merge.

File: docs/modules/ROOT/pages/native-memory.adoc [hunk 1]
```diff
@@ -0,0 +1,51 @@
+= Native Memory
+:description: Native Memory Store is designed as a pluggable memory manager which enables multiple memory stores for different data structures. These memory stores are all accessible by a common access layer that scales up to massive amounts of the main memory on a single JVM by minimizing the GC pressure. Native Memory Store enables predictable application scaling and boosts performance and latency while minimizing garbage collection pauses.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:2]:
```suggestion
:description: The Native Memory Store is a pluggable memory manager. It allows you to set up different configurations, or stores, for your application’s data structures on a single JVM. These memory stores form part of a common access layer which scales up to use a configured amount of memory on the JVM.
```

File: docs/modules/ROOT/pages/native-memory.adoc [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+{description}
+
+This topic explains how to configure native memory. It also brings together example of how you can configure your data structure with Native Memory.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:6]:
```suggestion
As well as providing more predictable application scaling, using a Native Memory Store boosts performance, reduces latency, and minimizes garbage collection pauses. 
```
If we change the title, I don't think the original sentence is needed.

File: docs/modules/ROOT/pages/native-memory.adoc [hunk 3]
```diff
@@ -0,0 +1,51 @@
+= Native Memory
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:1]:
```suggestion
= Configuring the Native Memory Store
```

File: docs/modules/ROOT/pages/native-memory.adoc [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+:description: Native Memory Store is designed as a pluggable memory manager which enables multiple memory stores for different data structures. These memory stores are all accessible by a common access layer that scales up to massive amounts of the main memory on a single JVM by minimizing the GC pressure. Native Memory Store enables predictable application scaling and boosts performance and latency while minimizing garbage collection pauses.
+
+{description}
+
+This topic explains how to configure native memory. It also brings together example of how you can configure your data structure with Native Memory.
+
+== Configuring Native Memory
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:8]:
```suggestion
== Configuring a Native Memory Store
```

File: docs/modules/ROOT/pages/native-memory.adoc [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+Below are the configuration options for the Native Memory feature.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:10]:
```suggestion
Below are the configuration options for the Native Memory Store.
```

File: docs/modules/ROOT/pages/native-memory.adoc [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+[cols="25%m,75%a"]
+|===
+|Field|Description
+
+|`allocatorType`
+| AllocatorType specifies one of 2 types of mechanism for allocating memory.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:17]:
```suggestion
| Specifies the mechanism for allocating memory; either XXXX or XXXX.
```
It makes sense to give both options if they are available, and to explain what they are used for.

File: docs/modules/ROOT/pages/native-memory.adoc [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+|`size`
+|Size of the total native memory to allocate.
+
+|`minBlockSize`
+|MinBlockSize is the size of smallest block that will be allocated.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:23]:
```suggestion
|Size of smallest block of memory to allocate.
```

File: docs/modules/ROOT/pages/native-memory.adoc [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+|`pageSize`
+|PageSize is the size of the page in bytes to allocate memory as a block.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:26]:
```suggestion
|Size of the page in bytes to allocate as a block of memory.
```

File: docs/modules/ROOT/pages/native-memory.adoc [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+|`metadataSpacePercentage`
+|MetadataSpacePercentage defines percentage of the allocated native memory that is used for the metadata of other map components such as index (for predicates), offset, etc.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:29]:
```suggestion
|Percentage of the allocated native memory that is used for the metadata of other map components such as index (for predicates), offset, etc.
```
What if the data structure isn't a map?

File: docs/modules/ROOT/pages/native-memory.adoc [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+|===
+
+=== Example Configuration
+
+The example configuration does the following:
+
+- Creates cluster with Native Memory and `POOLED` allocator enabled.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:37]:
```suggestion
- Creates a cluster with `Native Memory` and the `POOLED` allocator enabled.
```

File: docs/modules/ROOT/pages/native-memory.adoc [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+- Configures Map for that cluster that is using Native Memory.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:38]:
```suggestion
- Configures a map on the cluster to use native memory.
```

**Review by angelasimms — CHANGES_REQUESTED** [review]
I'm not sure about the introduction/description as I read had to read it a few times before I understood the advantages of a native memory store. I tried to reword but not sure if it is accurate.

File: docs/modules/ROOT/pages/native-memory.adoc [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+== Configuring a Native Memory Store
+
+Below are the configuration options for the Native Memory Store.
+
+[cols="25%m,75%a"]
+|===
+|Field|Description
+
+|`allocatorType`
+| Specifies the mechanism for allocating memory; either XXXX or XXXX.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/native-memory.adoc:17]: For this line, you need to add in the name of the two allocator types. It looks like the first is `POOLED` but I'm not sure about the second one.

**Review by angelasimms — CHANGES_REQUESTED** [review]

**Pull request merged**